### ISSUE TITLE
Positional Gradient vs "Regular" Geometric Gradient

### ISSUE THREAD

**Issue by saikrn112** [KByrski/RaySplatting#1]
Hi all, 
Thanks for releasing an amazing code on raytracing! It's exciting to see that everything is handwritten and in CUDA! 
Based on the paper, I am confused what the difference between positional gradient and geometric gradient that the paper is referring to. From the looks of the code, kernel is still taking norm of gradient along the means. Can you please clarify, what the difference between them is? 

![Image](https://github.com/user-attachments/assets/64d82455-12f6-48d3-b5ce-16ac470c40a3)

Additionally, I didnt see any FPS results in the paper. Since everything is now in CUDA including the optimization, I was hoping to see good speed ups in training. It will be really cool to see how fully CUDA code vs pytorch autograd will look like!

**Comment by KByrski** [comment]
Dear Mr. Kiran,

thank You for Your questions and valuable suggestions. As for the
difference, as the name suggests, the positional gradients (like in the GS)
are computed after projection in screen space, while ours not. As far as
the FPS are concerned on the other hand, so far we haven't conducted
experiments since we're still working on optimization. After rebuttal we
will update the paper.

Best regards

K. Byrski

pon., 3 lut 2025, 17:39 użytkownik Ramana Kiran ***@***.***>
napisał:

**Comment by KByrski** [comment]
The original 3DGS paper will provide You with more details. I suppose, they compute the positional gradients in the manner similar to this:

m' = m + lr_m * d_m
m_proj = project m to screen (for current train image)
m'_proj = project m' to screen (for current train image)
m_positional_gradient = || m'_proj - m_proj ||

**Comment by saikrn112** [comment]
I see the difference now. Thanks for the clarification! One of them is projected to the viewspace while the other is not.
